crates/glaredb/src/args/server.rs

```diff
@@ -75,12 +84,28 @@ pub struct ServerArgs {
     ///
     /// Enables Cloud to query glaredb without the overhead of managing system
     /// credentials or using a proxy.
-    #[clap(long, hide = true, value_parser)]
+    #[arg(long, hide = true, value_parser)]
     pub enable_simple_query_rpc: bool,
 
     /// API key for segment.
     ///
     /// (Internal)
-    #[clap(long, hide = true, value_parser)]
+    #[arg(long, hide = true, value_parser)]
     pub segment_key: Option<String>,
+    /// Enable the experimental Flight SQL API.
+    ///
+    /// This starts up a Flight SQL compatible API that can be used in place of
+    /// the postgres interface. This is an experimental feature and is subject to
+    /// change.
+    ///
+    /// By default, the Flight SQL API will be enabled on port 6789.
+    /// Use the `--flight-bind` option to change the port.
+    #[arg(long, default_value="false", action = clap::ArgAction::SetTrue)]
+    pub enable_flight_api: bool,
+
+    /// Disable the default postgres api.
+    ///
+    /// This will fully disable the postgres server on port 6543.
+    #[arg(long, default_value="false", action = clap::ArgAction::SetTrue)]
+    pub disable_postgres_api: bool,
```

Comment:
Why is this needed?

Reply:
i don't think it's needed, but it is subjectively useful. It takes up resources spinning up the postgres api, if users have no need for it, i feel like they should have a way to disable it.